test/fork/integration.balancerv2.test.js

```diff
@@ -0,0 +1,383 @@
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+const { getERC20, increaseTime, getHolderForTokenAddress, getTokenName, getTokenAddress, getHolderForToken } = require('../utils/test-helpers');
+const { ADDRESS_ZERO, ONE_DAY_IN_SECONDS } = require('../../lib/constants');
+const { impersonateAddress } = require('../../lib/rpc');
+const { eth } = require('../../lib/helpers');
+const { deployments } = require('hardhat');
+const { deploy } = deployments;
+
+const WETH = getTokenAddress('WETH');
+const NFT_URI = 'https://babylon.mypinata.cloud/ipfs/QmcL826qNckBzEk2P11w4GQrrQFwGvR6XmUCuQgBX9ck1v';
+const NFT_SEED = '504592746';
+
+const BALANCER_VAULT = '0xBA12222222228d8Ba445958a75a0704d566BF2C8';
+
+let weightedPool3Token;
+
+const BALANCER_POOLS = {
+  'USDC-DAI-USDT StablePool': {
+    poolAddress: '0x06Df3b2bbB68adc8B0e302443692037ED9f91b42',
+  },
+  'USDC-WETH WeightedPool2Token': {
+    poolAddress: '0x96646936b91d6B9D7D0c47C496AfBF3D6ec7B6f8',
+  },
+  'wstETH-ETH MetaStablePool': {
+    poolAddress: '0x32296969ef14eb0c6d29669c550d4a0449130230',
+  },
+  'wBTC-BADGER WeightedPool': {
+    poolAddress: '0xb460DAa847c45f1C4a41cb05BFB3b51c92e41B36',
+  },
+  'AAVE-WETH-USDT WeightedPool': {
+    poolAddress: () => weightedPool3Token, // variable will be set by before() task
```

Comment:
This is set later because we create our own pool with 3 tokens. We couldn't find any suitable balancer `WeightedPool`s with more than two tokens to test with.

Reply:
this looks great. 

Is there anyway to fetch all the balancer pools above for example $1M?

We would need a way to get the list so then we can offer it in the frontend.